src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacet.ts

```diff
@@ -254,7 +254,27 @@ export class DynamicHierarchicalFacet extends Component implements IDynamicHiera
      *
      * @examples electronics, electronics\,laptops
      */
-    basePath: ComponentOptions.buildListOption<string>({ defaultValue: [] }),
+    basePath: ComponentOptions.buildListOption<string>({ defaultValue: [], section: 'CommonOptions' }),
+
+    /**
+     * Whether to use the [`basePath`]{@link DynamicHierarchicalFacet.options.basePath} option as a filter for the results.
+     *
+     * **Note:**
+     * Setting this option to `true` and using the [`dependsOn`]{@link DynamicHierarchicalFacet.options.dependsOn} option can cause unexpected behaviour.
+     */
+    basePathShouldFilterResults: ComponentOptions.buildBooleanOption({
+      defaultValue: false,
```

Comment:
Should the default value be `true` to match the search api default documented on line 206 of `FacetRequest.ts`?

Reply:
Because it made more sense to have it false by default, even if for the API the default is true. The filtering part is an extra that wasn't part of the CategoryFacet.
I think if we see it from a support perspective, we will get less cases/questions if we don't filter by default.

Reply:
Hmm, I'm wondering why the searchapi chose `true`, since theoretically someone could use the api directly? Looking at @olamothe's comment above:

> the option is undefined and the Coveo platform will decide what will happen

It sounds like the behaviour change will take place on older JSUI versions not containing this PR. Could the support cases come anyway? Also, if people on existing versions upgrade to the new one, won't that be a behaviour change that could create confusion?


Personally, I feel it's more natural that a user would want to see results filtered on the base path as the default behaviour rather than the reverse. Aside from support cases, what other other reasons to deactivate by default?

Reply:
BasePath is new to this version and the older CategoryFacet component's basePath doesn't filter results.
Might also have found a bug with selected facet values not being affected by the basePath filter correctly. if thats the case, I will remove the option temporarily.